src/lib/es2015.reflect.d.ts

```diff
@@ -1,7 +1,7 @@
 declare namespace Reflect {
     function apply(target: Function, thisArgument: any, argumentsList: ArrayLike<any>): any;
     function construct(target: Function, argumentsList: ArrayLike<any>, newTarget?: any): any;
-    function defineProperty(target: object, propertyKey: PropertyKey, attributes: PropertyDescriptor): boolean;
+    function defineProperty(target: object, propertyKey: PropertyKey, attributes: PropertyDescriptor & ThisType<any>): boolean;

```

Comment:
Why not

```ts
function defineProperty<T extends object>(
  target: T,
  propertyKey: PropertyKey,
  attributes: PropertyDescriptor & ThisType<T>
)
```

Reply:
It doesn't make any sense, just try this case:
```ts
const obj = {a: 1};
defineProperty(obj, 'b', {
  get() {
    return this.b;
  }
})
```
The generics of typescript are completely static, they can\`t find the dynamic properties.

Reply:
Should I mark this as a resolved conversation? @Jessidhia